src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -76,14 +85,15 @@ public string UserName
         /// </devdoc>
         public string Password
         {
-            get
-            {
-                return InternalGetPassword();
-            }
-            set
-            {
-                _password = value;
-            }
+            get { return _password; }
+            set { _password = value ?? string.Empty; }
```

Comment:
We should not be storing both the string and the SecureString.  I see two main options:
1. Just do what desktop does.  Only have the _securePassword field, and have Password be a wrapper around it: setting the password marshals it to a SecureString, and getting it marshals it from the SecureString. (This unfortunately goes against the design of SecureString and what it's intended to do, but we're stuck with the behavior based on the design of this type.)
2. Have a single _password field of type object, and use it to store either the string or the SecureString, so that we only have one, and so that we don't unnecessarily force marshaling between the two on every access.

I suggest we do (2).  While (1) better matches desktop's behavior, it also adds non-trivial expense for the 99.9% case, and the case that already exists in .NET Core, that of using Password: (1) would force all getting/setting of password to roundtrip through a SecureString, which wouldn't add value but would add expense.

(2) would look something like this:
```C#
private object _password;

public string Password
{
    get
    {
        SecureString s = _password as SecureString;
        if (s != null) return MarshalToString(s);
        return (string)_password ?? string.Empty;
    }
    set
    {
        SecureString old = _password as SecureString;
        _password = value;
        old?.Dispose();
    }
}

public SecureString SecurePassword
{
    get
    {
        string p = _password as string;
        if (p != null) return MarshalToSecureString(p);
        SecureString s = _password as SecureString;
        return s != null ? s.Copy() : new SecureString();
    }
    set
    {
        SecureString old = _password as SecureString;
        _password = value?.Copy();
        old?.Dispose();
    }
}
```